src/client/App.jsx

```diff
@@ -1,60 +1,92 @@
 import React, { useState, useEffect } from 'react';
-import { Routes, Route } from 'react-router-dom';
+import { Routes, Route, useNavigate } from 'react-router-dom';
 
 import { Homepage } from './components/Homepage';
 import { Welcome } from './components/Welcome';
 import { Header } from './components/Header';
-
 import { Competition } from './components/Competition';
 import { Season } from './components/Season';
 
+import { API_URL, HTTP_METHOD, LOCAL_STORAGE } from './config';
+
 import './App.css';
 
 const App = () => {
-    const [isLoggedIn, setIsLoggedIn] = useState(true);
-    const [user, setUser] = useState("nathan");
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+    const [user, setUser] = useState(null);
+
+    const navigate = useNavigate();
 
-    const findUser = () => {
-        setIsLoggedIn(true);
-    };
+    useEffect(() => {
+        localStorage.getItem(LOCAL_STORAGE.TOKEN) ? setIsLoggedIn(true) : setIsLoggedIn(false);
+    }, []);
 
     useEffect(() => {
-        findUser();
-    });
+        if (!isLoggedIn) return;
+
+        const authenticateUser = async () => {
+            try {
+                const response = await fetch(API_URL.GET, {
+                    method: HTTP_METHOD.GET,
+                    headers: {
+                        Authorization: localStorage.getItem(LOCAL_STORAGE.TOKEN),
+                    },
+                });
+
+                const result = await response.json();
+
+                if (result) {
+                    setUser(result.data.username);
```

Comment:
You might also think about checking that result has a data property, since you're relying on it to get the username:

```js
if (result && result.data) {
```